openhands/server/session/agent.py

```diff
@@ -63,9 +66,14 @@ async def start(
             raise RuntimeError(
                 'Session already started. You need to close this session and start a new one.'
             )
-        await self._create_security_analyzer(config.security.security_analyzer)
-        await self._create_runtime(runtime_name, config, agent)
-        await self._create_controller(
+
+        self.loop = asyncio.new_event_loop()
+        self.thread = Thread(target=self._run, daemon=True)
+        self.thread.start()
```

Comment:
Each thread runs a runloop that will run all the async code without locking up the server.